realm/realm-library/src/main/java/io/realm/mongodb/sync/SubscriptionSet.java

```diff
@@ -0,0 +1,231 @@
+ * {@link #waitForSynchronization()} and its variants.
+ *
+ */
+@ObjectServer
+@Beta
+public interface SubscriptionSet extends Iterable<Subscription> {
+
+    /**
+     * The possible states a subscription set can be in
+     * TODO These are internal states, simplify public ones before release?
+     */
+    public enum State {
+        UNCOMMITTED(OsSubscriptionSet.STATE_VALUE_UNCOMMITTED),
+        PENDING(OsSubscriptionSet.STATE_VALUE_PENDING),
+        BOOTSTRAPPING(OsSubscriptionSet.STATE_VALUE_BOOTSTRAPPING),
+        COMPLETE(OsSubscriptionSet.STATE_VALUE_COMPLETE),
+        ERROR(OsSubscriptionSet.STATE_VALUE_ERROR),
+        SUPERCEDED(OsSubscriptionSet.STATE_VALUE_SUPERCEDED);
+
+        private final byte value;
+
+        State(byte nativeValue) {
+            this.value = nativeValue;
+        }
+
+        public static State fromNativeValue(long value) {
+            for (State state : values()) {
+                if (state.value == value) {
+                    return state;
+                }
+            }
+
+            throw new IllegalArgumentException("Unknown SubscriptionSetState code: " + value);
+        }
+    }
+
+    /**
+     * Find the first subscription that matches the given query.
+     *
+     * @param query query to search for.
+     * @return the subscription matching the query or {@code null} if no match was found.
+     */
+    @Nullable
+    Subscription find(RealmQuery query);
+
+    /**
+     * Find the subscription with a given name.
+     *
+     * @param name name of subscription to search for.
+     * @return the matching subscription or {@code null} if no subscription with that name was found.
+     */
+    @Nullable
+    Subscription find(String name);
+
+    /**
+     * Returns the current state of the SubscriptionSet. See {@link SubscriptionSet.State} for more
+     * details about each state.
+     *
+     * @return current state of the SubscriptionSet.
+     */
+    State getState();
+
+    /**
+     * Returns how many subscriptions are curently in this subscription set.
+     *
+     * @return the number of of subscriptions in the subscription set.
+     */
+    int size();
+
+    /**
+     * If {@link #getState()} returns {@link State#ERROR}, this method will return the reason.
+     * Errors can be fixed by modifying the subscription accordingly and then calling
+     * {@link #waitForSynchronization()}.
+     *
+     * @return the underlying error if the subscription set is in the {@link State#ERROR} state. For
+     * all other states {@code null} will be returned.
+     */
+    @Nullable
+    String getErrorMessage();
+
+    /**
+     * Wait for the subscription set to synchronize with the server. It will return when the
+     * server either accepts the set of queries and have downloaded data for them, or if an
+     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
+     * or wait for change listeners to trigger to see the downloaded data.
```

Comment:
```suggestion
     * Wait for the subscription set to synchronize with the server. It will return when the
     * server either accepts the set of queries and has downloaded data for them or if an
     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
     * or wait for change listeners to trigger to see the downloaded data.
```